src/sundials/sundials_context.c

```diff
@@ -30,30 +30,44 @@
 void sunAdiakCollectMetadata();
 #endif
 
-int SUNContext_Create(void* comm, SUNContext* sunctx)
+int SUNContext_Create(SUNComm comm, SUNContext* sunctx)
 {
   SUNProfiler profiler = NULL;
   SUNLogger logger     = NULL;
+
+  /* Since this function used to take a void* comm that was NULL 
+     when the comm was to be ignored, we check if its NULL here
+     and translate it to SUN_COMM_NULL to make the transition 
+     easier for users. */
+  if (!comm) 
+  {
+    comm = SUN_COMM_NULL;
+  }
```

Comment:
I'm not sure if we should correct this for the user, the compiler will likely warn them the input they're providing is incorrect